lib/gnat.ex

```diff
@@ -149,17 +169,29 @@ defmodule Gnat do
         {:reply, :ok, state}
     end
   end
+  def handle_call({:ping, pinger}, _from, state) do
+    :ok = :gen_tcp.send(state.tcp, "PING\r\n")
+    {:reply, :ok, Map.put(state, :pinger, pinger)}
+  end
 
-  defp perform_handshake(tcp) do
+  defp perform_handshake(tcp, connection_settings) do
     receive do
       {:tcp, ^tcp, operation} ->
-        "INFO" = operation |> String.split |> List.first |> String.upcase
-        :gen_tcp.send(tcp, "CONNECT {\"verbose\": false}\r\n")
+        {_, [{:info, options}]} = Parser.parse(Parser.new, operation)
```

Comment:
As far as I know, there's no guarantee that the entire INFO message will fit in this first TCP packet, so I think moving the message parsing into Parser is a good start, but I think it might be better to let the genserver receive packets until the parser finds the INFO message. And then maybe we wait for a message from the genserver in `init` to ensure the handshake completed successfully?